pkg/controlplane/controller/clusterauthenticationtrust/cluster_authentication_trust_controller.go

```diff
@@ -448,6 +454,44 @@ func filterExpiredCerts(certs ...*x509.Certificate) []*x509.Certificate {
 	return validCerts
 }
 
+// extractAPIServerOwnedFields extracts the configMap.data key-value pairs that
+// were set by the kube-apiserver manager in an Update operation. These are the
+// fields set by the ClusterAuthenticationTrust controller.
+func extractAPIServerOwnedFields(cm *corev1.ConfigMap) (map[string]string, error) {
+	var managedDataKeys []string
+
+	for _, f := range cm.ManagedFields {
+		if f.Manager != clusterAuthnTrustControllerName || f.Operation != metav1.ManagedFieldsOperationUpdate || f.FieldsType != "FieldsV1" {
+			continue
+		}
+		managedUnstructured := map[string]interface{}{}
+		if err := json.Unmarshal(f.FieldsV1.Raw, &managedUnstructured); err != nil {
+			return nil, fmt.Errorf("failed to unmarshal managed fields: %w", err)
+		}
+
+		dataMap, found, err := unstructured.NestedMap(managedUnstructured, "f:data")
```

Comment:
hmm... parsing out field manager seems significantly more complicated and fragile than tolerating / preserving unknown keys in the configmap